connector/protobuf/src/main/scala/org/apache/spark/sql/protobuf/utils/SchemaConverters.scala

```diff
@@ -230,4 +235,11 @@ object SchemaConverters extends Logging {
       case dt => StructField(fd.getName, dt, nullable = !fd.isRequired)
     }
   }
+
+  // Insert a dummy column to retain the empty message because
+  // spark doesn't allow empty struct type.
+  private def convertEmptyProtoToStructWithDummyField(desc: String): StructType = {
+    log.info(s"Keep $desc which is empty struct by inserting a dummy field.")
+    StructType(StructField("__dummy_field_in_empty_struct", StringType) :: Nil)
```

Comment:
minor: Shall we add "nullable = true" though that is the default?

Reply:
This will result in compiler warning. I believe they won't flip the default of this field, which is super risky.

Reply:
I see. We have set that in other areas. You can resolve this comment.